### ISSUE TITLE
Optimize Variable Handling to Remove Viper Dependencies

### ISSUE THREAD

**Issue by emattiza** [equinix-labs/otel-cli#83]
Per #82 , remove viper as a dep and handle env-vars and config through json unmarshaling. Should greatly improve performance and binary size.

**Comment by tobert** [comment]
Check out the Unmarshaler I added to cmd/config.go. It defaults the Config struct with otel-cli defaults automatically. I needed it for the testing stuff and if need be we can relocate it there but for now this seemed okay and useful. Please lmk.

I'm not sure if it makes things easier or harder, I guess it depends on the loading order and config should prob be first and overridden by envvars & flags so this hopefully helps.

**Comment by tobert** [comment]
@emattiza  #117 fixes Viper to actually work. As far as I can tell the envvars it supported never worked, and only those supported by the SDK (e.g. `OTEL_EXPORTER_OTLP_ENDPOINT`) worked. So even OTEL_CLI_ATTRIBUTES didn't. As a result, the viper config file functionality likely never worked and besides isn't documented, and would need work to be better than the JSON.

I almost ripped viper out while troubleshooting, but ultimately figured it out. I still don't like how the config file looks in viper because of how flags are named to be flags and not config keys. 

Importantly, in my branch the integration tests now validates envvars are working as expected. So we should make the change now. I'm exploring what the code would look like v.s. the viper fix in a local branch.

**Comment by tobert** [comment]
@emattiza I tried it https://github.com/tobert/otel-cli/commit/e80819bea6a0279cf9282e4c78222e9c0e5de677

**Comment by tobert** [comment]
Indeed, this does reduce binary size noticeably! About 1.2MB on my x86_64 machine. I still need to add json config parsing back but I don't expect that to add more than a couple kilobytes tops.

Before (`109-fix-envvar-settings` branch):
```
atobey@usagi:~/src/otel-cli$ ls -l otel-cli
-rwxr-xr-x 1 atobey atobey 18043798 Jan  9 20:12 otel-cli
```

After (`experiment-remove-viper` branch):
```
atobey@usagi:~/src/otel-cli$ ls -l otel-cli 
-rwxr-xr-x 1 atobey atobey 16871993 Jan  9 19:49 otel-cli
```
